### DISCUSSION TITLE
KeyError: 'slurm' when pre-processing

### DISCUSSION THREAD

**Discussion by kexinliu0001** [nextpyp/.github#30] [Q&A] [answered]
hello,

I'm trying the command line interface PYP.

I'm using Standalone mode rather than slurm cluster. 

I noticed the note in installation step4 that "To run PYP in Standalone mode, make sure there [slurm] section in the configuration file is removed." and removed the section.
however, it still show KeyError: 'slurm' :
<img width="990" alt="image" src="https://github.com/nextpyp/.github/assets/110724494/e5c921a7-9d85-4007-a94b-f4dfec8e0216">
how can I solve it?

ps: I cannot find "/opt/pyp/bin/run/pyp" to see if there is something wrong. There is only "/opt/PYP" where I installed the PYP. And there is no bin folder in the PYP folder

**Comment by yezhou867** [comment]
Hi, that part is trying to find the singularity command path from your system.  If you have that Apptainer installed, you probably need to specify the path under the keyword [pyp]:
singularity ="/YOUR_PATH_TO_SINGULARITY"
So the code won't try to search under ["slurm"] when using the command line. We could change this in the future to avoid this conflict.

**Reply by kexinliu0001** [reply]
thank you!

I added the path
<img width="584" alt="image" src="https://github.com/nextpyp/.github/assets/110724494/ee9d09e2-1535-47bf-963d-83147bb27a7c">

however, it still gave the error
<img width="861" alt="image" src="https://github.com/nextpyp/.github/assets/110724494/f8cf0e2b-d6a3-4bdc-b307-25853f9e524a">

**Reply by yezhou867** [reply, marked as answer]
We haven't created a clear logic to run the command line in a standalone machine so far. Only the web interface supports the standalone option.  By the way, with limited resources (cpu cores, in your setting, availabeCpus=1), it could take much longer time to finish the refinement jobs.  
For the question 
> cannot find "/opt/pyp/bin/run/pyp". nextpyp runs inside the Apptainer, the command path will not show up outside of it.

**Reply by kexinliu0001** [reply]
thank you for your reply!
